File: java/darts/src/main/java/Darts.java
class Darts {
    private Dart dart;
    private Circle[] circles = {
        new Circle(0, 1, 10),
        new Circle(1, 5, 5),
        new Circle(5, 10, 1)
    };

    Darts(double x, double y) {
        this.dart = new Dart(x, y);
    }

    int score() {
        for (Circle circle : circles) {
            if (dart.hasLandedOn(circle)) {
                return circle.score();
            }
        }
        return 0;
     }
}

File: go/darts/darts.go
package darts

import "math"

// Circle represents a target circle in Darts game.
type Circle struct {
	radius float64
	score  int
}

var circles = [...]Circle{
	{radius: 1.0,  score: 10},
	{radius: 5.0,  score: 5},
	{radius: 10.0, score: 1},
}

// Score returns the earned points in Darts game.
func Score(x, y float64) int {
	dist := math.Sqrt(x*x + y*y)
	for _, circle := range circles {
		if dist <= circle.radius {
			return circle.score
		}
	}
	return 0
}
